meshtastic-mqtt-connect.py

```diff
@@ -606,6 +618,51 @@ def maybe_store_nodeinfo_in_db(info):
         db_connection.close()
 
 
+def maybe_store_position_in_db(node_id, position):
+    if debug:
+        print(f"Position report for: {get_short_name_by_id(node_id)}")
+        print(position)
+
+    # Must have at least a lat/lon
+    if position.latitude_i != 0 and position.longitude_i != 0:
+        # Convert from integer lat/lon format to decimal format.
+        latitude = position.latitude_i * 1e-7
+        longitude = position.longitude_i * 1e-7
+
+        # Get the best timestamp we can, starting with local time.
+        timestamp = time.gmtime()
+        # Then, try the timestamp from the position protobuf.
+        if position.timestamp > 0:
+             timestamp = time.gmtime(position.timestamp)
+        # Then, try the time from the position protobuf.
+        if position.time > 0:
+             timestamp = time.gmtime(position.time)
+        # Convert timestamp to datetime for database use
+        timestamp = datetime.fromtimestamp(mktime(timestamp))
+
+        try:
+            db_connection = sqlite3.connect(db_file_path)
+            db_cursor = db_connection.cursor()
+
+            # Check for an existing entry for the timestamp; this indicates a position that has bounced around the mesh.
+            existing_record = db_cursor.execute('SELECT * FROM positions WHERE node_id=? AND timestamp=?', (node_id,timestamp)).fetchone()
```

Comment:
any reason to not use the message id to check for duplicates?  I'm not sure it matters but two nodes could report at the same time.

Reply:
I suppose you'd need to add message id to this table in the db for that to work...

Reply:
This checks for a position send from the same node at the same time, which we can always ignore, even if it isn't the same message ID, so I think this might be more robust (and don't message IDs sometime change inside the network?  Not sure).

Reply:
sounds good.